server.js

```diff
@@ -3,36 +3,102 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import topMusicData from './data/top-music.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
-
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const port = process.env.PORT || 9000
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+//Model to access the DBase
+const Music = new mongoose.model('Music',{
+  id: Number,
+  trackName: String,
+  artistName: String,
+  genre: String,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number
+})
+
+//Custom enviormental Variable    ---    RESET_DATABASE=true npm run dev
+if (process.env.RESET_DATABASE) {
+  console.log('Resetting Database!')
+  // Function to populate the DBase 
+  const populateDataBase = async () => {
+    await Music.deleteMany({})
+
+    topMusicData.forEach(item => {
+      const newMusic = new Music(item)
+      newMusic.save()
+    })
+  }
+  populateDataBase()
+}
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Welcome to the Songs data API')
+})
+
+// All info from JSON
+app.get('/topsongs', async (req, res) => {
+  const allMusic = await Music.find()
+  res.json(allMusic)
+})
+
+//Searching with conditionals
+app.get('/topsongs/tracks', async (req, res) => {
+  const bpmQuery = parseInt(req.query.bpm)
+  const lengthQuery = parseInt(req.query.length)
+  const popularityQuery = parseInt(req.query.popularity)
+  let tracks = await Music.find()
+
+  if (bpmQuery) {
+    tracks = await Music.find({ 'bpm': bpmQuery })
+  } else if (lengthQuery){
+    tracks = await Music.find({ 'length': lengthQuery })
+  } else if (popularityQuery) {
+    tracks = await Music.find({ 'popularity': popularityQuery })
+  }
+  else {
+    res.status(404).json({ error: 'Sorry. Track not found'})
+  }
```

Comment:
Very nice that you worked with conditionals here. Very clean and easy to understand. Also cool that you used parseInt.